msteams-platform/concepts/authentication/nested-authentication.md

```diff
@@ -0,0 +1,343 @@
+
+```http
+brk-multihub://<your_domain>
+```
+
+Your domain must include only the origin and not its subpaths. For example:
+
+✔️ brk-multihub://myapp.teams.microsoft.com <br>
+❌ brk-multihub://myapp.teams.microsoft.com/go
+
+### Initialize public client app
+
+> [!NOTE]
+> To ensure successful authentication, initialize TeamsJS before you initialize MSAL.
+
+Initialize MSAL and get an instance of the public client app to get access tokens, when needed.
+
+# [JavaScript](#tab/js1)
+
+```javascript
+import {
+  AccountInfo,
+  IPublicClientApplication,
+  createNestablePublicClientApplication,
+} from "@azure/msal-browser";
+
+const msalConfig = {
+  auth: {
+    clientId: "your_client_id",
+    authority: "https://login.microsoftonline.com/{your_tenant_id}",
+  },
+};
+
+let pca: IPublicClientApplication;
+
+export function initializePublicClient() {
+  console.log("Starting initializePublicClient");
+  return createNestablePublicClientApplication(msalConfig).then(
+    (result) => {
+      console.log("Client app created");
+      pca = result;
+      return pca;
+    }
+  );
+}
+```
+<!--
+# [C#](#tab/cs1)
+
+```csharp
+using Microsoft.Identity.Client;
+using System;
+using System.Threading.Tasks;
+
+public class MsalClient
+{
+    private static IPublicClientApplication pca;
+
+    public static async Task<IPublicClientApplication> InitializePublicClientAsync()
+    {
+        Console.WriteLine("Starting initializePublicClient");
+
+        var msalConfig = PublicClientApplicationBuilder.Create("your_client_id")
+            .WithAuthority("https://login.microsoftonline.com/{your_tenant_id}")
+            .Build();
+
+        pca = msalConfig;
+        Console.WriteLine("Client app created");
+
+        return pca;
+    }
+}
+```
+-->
+---
+
+### Acquire your first token
+
+The tokens acquired by MSAL.js through nested app authentication are issued for your Microsoft Entra app registration ID. MSAL.js handles token acquisition for user authentication. It tries to get an access token silently. If that's unsuccessful, it prompts the user for consent. The token is then used to call the Microsoft Graph API or other Microsoft Entra ID protected resources. Unlike the OBO flow, you don't need to preauthorize your hosts to call the endpoints.
+
+To acquire a token, follow these steps:
+
+1. Use MSAL.js to acquire tokens for your app ID. For more information, see [acquire and use an access token](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/acquire-token.md).
+
+1. Use `getActiveAccount` API to verify if there's an active account to call the `publicClientApplication`. If there's no active account, try to retrieve one from the cache with `getAccount`, using additional filter parameters, such as `tenantID`, `homeAccountId`, and `loginHint` from [Context interface](../../tabs/how-to/using-teams-client-library.md#updates-to-the-context-interface).
```

Comment:
Can we point to the MSAL cacheing documentation here, like: 

"If there's no active account, try to retrieve one from the cache with `getAccount`, using additional filter parameters, such as `tenantID`, `homeAccountId`, and `loginHint` from [Context interface](../../tabs/how-to/using-teams-client-library.md#updates-to-the-context-interface). _Documentation for enabling the cache in MSAL can be found here: [microsoft-authentication-library-for-js/lib/msal-browser/docs/caching.md at dev · AzureAD/microsoft-authentication-library-for-js (github.com)](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/caching.md)_

Reply:
@chrisyanasak @baljesingh When I select the link for Context Interface, the page that opens states that - There isn't anything to compare. Please confirm if the link is correct.

<img width="878" alt="image" src="https://github.com/user-attachments/assets/03f1da90-17fe-4c4b-80f7-6304d3cd4810">

Reply:
@chrisyanasak I added the link to the MSAL documentation for enabling cache in the See also section.